src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+            // requirement.MessageAuthenticationAuditLevel = sessionProtocolFactory.MessageAuthenticationAuditLevel;
+            requirement.Properties[ServiceModelSecurityTokenRequirement.MessageDirectionProperty] = MessageDirection.Input;
+            if (sessionProtocolFactory.EndpointFilterTable != null)
+            {
+                requirement.Properties[ServiceModelSecurityTokenRequirement.EndpointFilterTableProperty] = sessionProtocolFactory.EndpointFilterTable;
+            }
+            this.sessionTokenAuthenticator = this.sessionProtocolFactory.SecurityTokenManager.CreateSecurityTokenAuthenticator(requirement, out this.sessionTokenResolver);
+            if (!(this.sessionTokenAuthenticator is IIssuanceSecurityTokenAuthenticator))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.SecuritySessionRequiresIssuanceAuthenticator, typeof(IIssuanceSecurityTokenAuthenticator), this.sessionTokenAuthenticator.GetType())));
+            }
+            if (sessionTokenResolver == null || (!(sessionTokenResolver is ISecurityContextSecurityTokenCache)))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.SecuritySessionRequiresSecurityContextTokenCache, this.sessionTokenResolver.GetType(), typeof(ISecurityContextSecurityTokenCache))));
+            }
+            this.sessionTokenCache = (ISecurityContextSecurityTokenCache)this.sessionTokenResolver;
+            return Task.CompletedTask;
+        }
+
+        public void StopAcceptingNewWork()
+        {
+            this.acceptNewWork = false;
+        }
+
+        int GetPendingSessionCount()
+        {
+            return this.pendingSessions1.Count + this.pendingSessions2.Count;
+            //+ ((IInputQueueChannelAcceptor)this.channelAcceptor).PendingCount;
+        }
+
+        void AbortPendingChannels(CancellationToken token)
+        {
+            ClosePendingChannels(token);
+        }
+
+        async void ClosePendingChannels(CancellationToken token)
+        {
+            var tasks = new Task[this.activeSessions.Count];
+            lock (ThisLock)
+            {
+                int index = 0;
+                if (typeof(IReplyChannel).Equals(this.AcceptorChannelType))
+                {
+                    foreach (ServerSecuritySimplexSessionChannel securitySessionSimplexChannel in this.activeSessions.Values)
+                    {
+                        tasks[index] = securitySessionSimplexChannel.CloseAsync(token);
+                        index++;
+                    }
+                }
+            }
+            await Task.WhenAll(tasks);
+        }
+
+        void ConfigureSessionSecurityProtocolFactory()
+        {
+            //TODO while implementing message security
+
+            //if (this.sessionProtocolFactory is SessionSymmetricMessageSecurityProtocolFactory)
```

Comment:
Remove commented code. It's going to be a long time before we can do full message security.

Reply:
Can I keep it ? I would like to target Message Security in 2021 first quarter(provided it's feasible)... otherwise let me know I can remove..